utils/parallelisation/parallelisation.go

```diff
@@ -292,6 +293,39 @@ func RunActionWithParallelCheck(ctx context.Context, action ActionFunc, checkAct
 	return
 }
 
+// RunPeriodicCheckWithAction continuously evaluates checkFunc and, when it returns true, executes actionFunc before waiting for pauseBetweenEvaluations.
+// The loop stops when the context is cancelled or when either the checkFunc or the actionFunc returns an error.
+func RunPeriodicCheckWithAction(ctx context.Context, checkFunc CheckFuncWithErr, actionFunc ActionFunc, pauseBetweenEvaluations time.Duration) error {
+	if checkFunc == nil {
+		return commonerrors.UndefinedVariable("check function")
+	}
+
+	if actionFunc == nil {
+		return commonerrors.UndefinedVariable("action function")
+	}
+
+	for {
+		err := DetermineContextError(ctx)
+		if err != nil {
+			return err
+		}
+
+		runAction, err := checkFunc(ctx)
+		if err != nil {
+			return err
+		}
+
+		if runAction {
+			err = actionFunc(ctx)
+			if err != nil {
+				return err
+			}
+		}
+
+		SleepWithContext(ctx, pauseBetweenEvaluations)
+	}
```

Comment:
```suggestion
type checkResult struct {
	run bool
	err error
}

// RunPeriodicCheckWithAction continuously evaluates checkFunc and, when it returns true, executes actionFunc before waiting for pauseBetweenEvaluations.
// The loop stops when the context is cancelled or when either the checkFunc or the actionFunc returns an error.
func RunPeriodicCheckWithAction(ctx context.Context, checkFunc CheckFuncWithErr, actionFunc ActionFunc, pauseBetweenEvaluations time.Duration) (err error) {
	_, _, err = RunActionWithParallelCheckAndResult(
		ctx,
		func(actionCtx context.Context) (_ error) {
			<-actionCtx.Done()
			return
		},
		func(checkCtx context.Context) (res checkResult, ok bool) {
			run, checkErr := checkFunc(checkCtx)
			res = checkResult{run: run, err: checkErr}
			ok = true
			return
		},
		func(res checkResult) (err error) {
			if res.err != nil {
				err = res.err
				return
			}
			if res.run {
				err = actionFunc(ctx)
				return
			}
			return
		},
		pauseBetweenEvaluations,
	)

	return
```

It might be slightly more complex but it reuses the existing generic function, but I understand if this is overkill